src/test/scala/mesosphere/marathon/api/akkahttp/DirectivesTest.scala

```diff
@@ -0,0 +1,44 @@
+package mesosphere.marathon
+package api.akkahttp
+
+import akka.http.scaladsl.model.Uri
+import akka.http.scaladsl.model.Uri.{ Path, Query }
+import akka.http.scaladsl.server.MalformedQueryParamRejection
+import akka.http.scaladsl.server.PathMatcher.{ Matched, Unmatched }
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import mesosphere.UnitTest
+import mesosphere.marathon.state.{ AppDefinition, PathId }
+import mesosphere.marathon.test.GroupCreation
+
+class DirectivesTest extends UnitTest with ScalatestRouteTest {
+  import Directives._
+
+  def objectName(obj: AnyRef) = obj.getClass.getName
+
+  val route = extractTaskKillingMode { mode =>
+    complete(objectName(mode))
+  }
+
+  "Directives" should {
+    "extract scale killing mode from request" in {
+      Get().withUri(Uri./.withQuery(Query("scale" -> "true"))) ~> route ~> check {
```

Comment:
Create tests! Is there a way to test the directive without `route` and `check`?

Reply:
I think there is, but since directives are highly coupled to routes (for example `apply` of the directive returns a `Route`), it will be less concise compared to current way of testing them.

Reply:
Btw I meant `Great tests`. Shame on me for this Freudian slip.